internal/handlers/generate_sbom_test.go

```diff
@@ -0,0 +1,88 @@
+package handlers
+
+import (
+	"context"
+	"encoding/json"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	storagev1alpha1 "github.com/rancher/sbombastic/api/storage/v1alpha1"
+	"github.com/rancher/sbombastic/internal/messaging"
+	"github.com/rancher/sbombastic/pkg/generated/clientset/versioned/scheme"
+	"github.com/spdx/tools-golang/spdx"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+func TestGenerateSBOMHandler_Handle(t *testing.T) {
+	image := &storagev1alpha1.Image{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-image",
+			Namespace: "default",
+		},
+		Spec: storagev1alpha1.ImageSpec{
+			ImageMetadata: storagev1alpha1.ImageMetadata{
+				Registry:    "docker",
+				RegistryURI: "docker.io",
+				Repository:  "busybox",
+				Tag:         "1.37.0",
+				Platform:    "linux/amd64",
+				Digest:      "sha256:123",
```

Comment:
The test is actually running trivy and building a real sbom of this image. Do I get it right?

Some questions:
- Should we point to an image we have control (this version of busybox is going to disappear from the docker registry at some point in time)
- Should we create right now an interface for the sbom generator? This would allow us to use a mock for trivy right now and later to have a grype based sbom generator

Reply:
I've tracked both as an issue.

The adapter is out of scope for the MVP, keep in mind that this will only move the integration test on the adapter side.

Reply:
I'm fine with that